lib/ssl-test.rb

```diff
@@ -83,8 +88,12 @@ def self.test_ocsp_revocation chain, open_timeout: 5, read_timeout: 5, redirecti
         return ocsp_soft_fail_return("Missing OCSP URI in authorityInfoAccess extension") unless ocsp
 
         ocsp_uri = URI(ocsp[/URI:(.*)/, 1])
-        http_response = follow_ocsp_redirects(ocsp_uri, request.to_der, open_timeout: open_timeout, read_timeout: read_timeout, redirection_limit: redirection_limit)
-        return ocsp_soft_fail_return("OCSP response request failed") unless http_response
+        http_response, ocsp_request_error = follow_ocsp_redirects(ocsp_uri, request.to_der, open_timeout: open_timeout, read_timeout: read_timeout, redirection_limit: redirection_limit)
+        if http_response.nil?
+          ocsp_request_error_return = ocsp_soft_fail_return("OCSP response request failed (OCSP URI: #{ocsp_uri}, error: #{ocsp_request_error})")
+          @ocsp_request_error_cache[unicity_key] = { error: ocsp_request_error_return, cache_until: Time.now + OCSP_REQUEST_ERROR_CACHE_DURATION }
+          return ocsp_request_error_return
+        end
```

Comment:
Looks nice, but I think we should also cache all errors between this and like 117 because if it fails after this but before success we won't cache anything and spam then endpoint too. Maybe you can extract this in the `ocsp_soft_fail_return` method or something like that?

Reply:
Yes you're right. Handled in 2e5cd14.
I didn't cache the steps before though, because I'm not sure we can't have a certificate with the information for the OCSP URI and one without for the same certificate ID.

If it's possible and we cache the error, it'd prevent us to reach the OCSP responder while it would have been possible. And anyway, we don't hit the OCSP responder at this point so it's not a big deal to not cache this part.

Reply:
Perfect! yeah we don't care about the first errors as they don't make any requests, it's cheap.